src/Trigger.cc

```diff
@@ -98,23 +98,51 @@ class TriggerTimer final : public Timer {
 	double time;
 };
 
-Trigger::Trigger(Expr* arg_cond, Stmt* arg_body,
-                 Stmt* arg_timeout_stmts,
-                 Expr* arg_timeout, Frame* arg_frame,
-                 bool arg_is_return, const Location* arg_location)
+Trigger::Trigger(const Expr* cond, Stmt* body, Stmt* timeout_stmts,
+                 Expr* timeout_expr, Frame* frame,
+                 bool is_return, const Location* location)
+	{
+	timeout_value = -1;
+
+	if ( timeout_expr )
+		{
+		ValPtr timeout_val;
+
+		try
+			{
+			timeout_val = timeout_expr->Eval(frame);
+			}
+		catch ( InterpreterException& )
+			{ /* Already reported */ }
+
+		timeout_value = timeout_val->AsInterval();
```

Comment:
Is `timeout_val` valid here if there was an exception? In the previous version it checked for null before trying to use it.

Reply:
Good catch, thanks.  Fix pushed.